dashboard/src/components/AppView/ResourceTable/ResourceItem/SecretItem/SecretItemDatum.v2.tsx

```diff
@@ -13,11 +16,29 @@ interface ISecretItemDatumProps {
 
 function SecretItemDatum({ name, value }: ISecretItemDatumProps) {
   const [hidden, setHidden] = React.useState(true);
+  const [copied, setCopied] = React.useState(false);
   const toggleDisplay = () => setHidden(!hidden);
+  const setCopiedTrue = () => {
+    setCopied(true);
+    setTimeout(() => {
+      setCopied(false);
+    }, 1000);
+  };
```

Comment:
This code may cause an error if the user navigates to a different page when the tooltip is showed. This component will be unmounted and the `setCopied` method will disappear from the context. To avoid this, you can use `useRef` to keep the timeout reference and `useEffect` to control it:

```suggestion
  const [copied, setCopied] = React.useState(false);
  const toggleDisplay = () => setHidden(!hidden);
  const copyTimeout = useRef(null);

  // You can now remove this intermediate method.
  const setCopiedTrue = () => {
    setCopied(true);
  };

  useEffect(() => {
    if (copied) {
      copyTimeout.current = setTimeout(() => {
        setCopied(false);
      }, 1000);
    }

    return () => {
      if (copyTimeout.current != null) {
        clearTimeout(copyTimeout.current);
      }
    }
  }, [copied])
```

Reply:
good catch!